### ISSUE TITLE
Completed Milestones stay in active overview

### ISSUE THREAD

**Issue by krrisis** [Giveth/giveth-dapp#833]
Seems this only happens with the Milestones without a cap? 

![image](https://user-images.githubusercontent.com/6533925/59461628-1b537f80-8e22-11e9-926b-925fdd6217f8.png)

**Comment by GriffGreen** [comment]
oh this is for the "Archive" work around that @ewingrj made... we need to move archive'd milestones to a different tab

**Comment by krrisis** [comment]
before these just moved to 'paid' which makes sense to me. Now they stay here, even when paid. as said, only seems to be with the ones that have no cap. 

It is prob bcs you can have multiple payouts when no cap, which I think is a feature we can/should drop, only recently heard that is even possible. I think we should drop it because it causes exactly this issue -> if you can have multiple rounds of payments these type of Milestones should neither live in 'archived' nor in paid, it makes categorization very hard & confusing for users. 
I think it's easier (our system is already so complex) that once you hit 'completed' a Milestone gets paid out and can no longer receive funds. Need more funds? Create a new Milestone. #KISSprinciple.

**Comment by Danibelle** [comment]
![image](https://user-images.githubusercontent.com/39466970/61236831-96000b00-a6fe-11e9-8091-653be80a8a92.png)

**Comment by arbreton** [comment]
If the milestone is completed it should have an Archive button

**Comment by arbreton** [comment]
The archive button actually transforms the milestone to the paid section *

**Comment by GriffGreen** [comment]
when a milestone without a reviewer can be collected multiple times... this is an issue

**Comment by aminlatifi** [comment]
Finally got it, it happens when an uncapped milestone with donations and without recipient is archived.
![image](https://user-images.githubusercontent.com/5684607/62539611-e46e7880-b86a-11e9-8b96-80aad7cb6151.png)

**Comment by aminlatifi** [comment]
@arbreton It think in line bellow, `gte` should be change by `gt`
https://github.com/Giveth/giveth-dapp/blob/e98dee88767e59457d1120160490085dc0e40959/src/components/ArchiveMilestoneButton.jsx#L18
As is here:
https://github.com/Giveth/giveth-dapp/blob/e98dee88767e59457d1120160490085dc0e40959/src/models/Milestone.js#L318-L322

**Comment by aminlatifi** [comment]
> Finally got it, it happens when an uncapped milestone with donations and without recipient is archived.
> ![image](https://user-images.githubusercontent.com/5684607/62539611-e46e7880-b86a-11e9-8b96-80aad7cb6151.png)

The other case is when user (Giveth0 here) is reviewer and recipient is not collected yet. It happens for both capped and uncapped milestones.
![image](https://user-images.githubusercontent.com/5684607/62599462-9062a300-b901-11e9-98b1-c84b722d967b.png)

**Comment by arbreton** [comment]
Yes exactly, those are the cases, basically any uncapped milestone if you are the receiver you will have an infinite active milestone and there is no way to remove it from there.

There is already and archive button / functionality implemented but is not working correctly it actually breaks things sometimes, what needs to happen is just mark milestones as archived and put the on the paid tab or even better into an 'Archived' Tab! 

Basically just to avoid clutter on your active tab, as you can see this can be a pretty hard task when you have various roles!

Keep me posted @aminlatifi !, feel free to ask anything

**Comment by arbreton** [comment]
> @arbreton It think in line bellow, `gte` should be change by `gt`
> https://github.com/Giveth/giveth-dapp/blob/e98dee88767e59457d1120160490085dc0e40959/src/components/ArchiveMilestoneButton.jsx#L18
> 
> 
> As is here:
> https://github.com/Giveth/giveth-dapp/blob/e98dee88767e59457d1120160490085dc0e40959/src/models/Milestone.js#L318-L322

It makes sense!, there is various inconsistencies the cleaner the code the better if it doesn't break anything that should be the way to go

**Comment by aminlatifi** [comment]
> Yes exactly, those are the cases, basically any uncapped milestone if you are the receiver you will have an infinite active milestone and there is no way to remove it from there.
> 
> There is already and archive button / functionality implemented but is not working correctly it actually breaks things sometimes, what needs to happen is just mark milestones as archived and put the on the paid tab or even better into an 'Archived' Tab!
> 
> Basically just to avoid clutter on your active tab, as you can see this can be a pretty hard task when you have various roles!
> 
> Keep me posted @aminlatifi !, feel free to ask anything

So, I think we should find those milestone which are archived and user has nothing to do with them (Collect, Disburse, or etc.), and place them on "Paid" or maybe "Archived" tab. If you agree, I prefer "Paid" tab since it seems the payments will paid to recipient at last and it cannot be cancelled!

**Comment by aminlatifi** [comment]
Also there is no "Set Recipient" button for completed milestone with no recipient. It should be added too.

**Comment by aminlatifi** [comment]
> Also there is no "Set Recipient" button for completed milestone with no recipient. It should be added too.

If it's against rules, we should prevent empty recipient at the creation or marking complete time.

**Comment by aminlatifi** [comment]
> Finally got it, it happens when an uncapped milestone with donations and without recipient is archived.
> ![image](https://user-images.githubusercontent.com/5684607/62539611-e46e7880-b86a-11e9-8b96-80aad7cb6151.png)

In case of capped milestone with reviewer, user cannot "Mark Complete" a milestone without recipient. It's similar to uncapped case. So we should prevent archiving when no recipient is set for milestone.

**Comment by aminlatifi** [comment]
Consider this uncapped milestone:
**Roles:**
_User0: Milestone Reviewer
User1: Campaign Owner and Recipient
User2: Milestone Owner_

**Scenario Steps:**
1. User2 creates milestone
2. User2 (or anyone) does donation to milestone
3. User2 archive the milestone

After above steps, just disburse/collect button is available for User2/User1. 
The problem is reviewer (User0) paid no role and clicking disburse or collect buttons show error message.
